packages/core/src/components/date-input/date-input.tsx

```diff
@@ -0,0 +1,491 @@
+    event: IxDatePickerReworkCustomEvent<DateChangeEventRework>
+  ) {
+    this._from = event.detail.from;
+    this._to = event.detail.to;
+
+    const ariaLabelFormat = 'dddd MMMM D, YYYY';
+    this.dropdownButtonAriaLabel = `Change Date,
+      ${this._from && dayjs(this._from, this.format).format(ariaLabelFormat)} -
+      ${this._to && dayjs(this._to, this.format).format(ariaLabelFormat)}`;
+
+    this.dateChange.emit({
+      from: this._from,
+      to: this._to,
+    });
+
+    this.updateValidity();
+  }
+
+  private readonly clear = () => {
+    this._from = undefined;
+    this._to = undefined;
+    this.firstInput.value = '';
+    this.secondInput.value = '';
+
+    this.firstInput.focus();
+    this.ixClear.emit();
+    this.updateValidity();
+  };
+
+  onFromInputChange(event) {
+    if (this._from !== event.target.value) this._from = event.target.value;
+    this.onInputChange(event, 'from');
+  }
+
+  onToInputChange(event) {
+    if (this._to !== event.target.value) this._to = event.target.value;
+    this.onInputChange(event, 'to');
+  }
+
+  onInputChange(event: InputEvent, input: string) {
+    this.ixChange.emit({
+      InputName: input,
+      Event: event,
+    });
+
+    this.updateValidity();
+  }
+
+  private updateValidity() {
+    if (this.wasValidated) {
+      this.setInputValidity();
+    }
+  }
+
+  private readonly setInputValidity = () => {
+    const param: DateValidatorParam = {
+      from: this.firstInput.value,
+      to: this.secondInput.value,
+      format: this.format,
+      min: this.minDate,
+      max: this.maxDate,
+    };
+
+    const isValid = this.validator.validate(param);
+    if (!isValid) {
+      this.firstInput.setCustomValidity(this.validator.errorMessage);
+      this.secondInput.setCustomValidity(this.validator.errorMessage);
+    } else {
+      this.firstInput.setCustomValidity('');
+      this.secondInput.setCustomValidity('');
+    }
+
+    this.dateInputDiv.classList.toggle('is-invalid', !isValid);
+    this.hostElement.classList.toggle('invalid', !isValid);
+    this.hostElement.classList.toggle('is-valid', isValid);
+
+    this.wasValidated = true;
+    this.errorMessage = this.validator.errorMessage;
+  };
+
+  componentWillLoad() {
+    this._from = this.from;
+    this._to = this.to;
+
+    const validatorEntries = convertInputValidators(this.validators).filter(
+      (v) => DATE_VALIDATORS.includes(v.name as ValidatorName)
+    );
+    this.validator = getValidator(validatorEntries);
+
+    const hiddenInput = document.createElement('input');
+    hiddenInput.style.display = 'none';
+    this.hostElement.appendChild(hiddenInput);
+
+    hiddenInput.form.addEventListener('submit', this.setInputValidity);
+  }
+
+  componentDidRender() {
+    const inputs = this.hostElement.shadowRoot.querySelectorAll('input');
+    inputs.forEach((input) => {
+      input.setAttribute(
+        'size',
+        input.getAttribute('placeholder').length as unknown as string
```

Comment:
I think this approach is fine but we should factor in the input padding so nothing gets cut off.